### ISSUE TITLE
Support UNNEST within projection

### ISSUE THREAD

**Issue by findepi** [prestosql/presto#6291]
This works:

```
presto> SELECT histogram(e)
     -> FROM (VALUES 'abccccdaee') t(x)
     -> LEFT JOIN UNNEST(regexp_extract_all(x, '.')) u(e) ON true;
           _col0
---------------------------
 {a=2, b=1, c=4, d=1, e=2}
```

So this could probably  be made to work too:
```
presto> SELECT (SELECT histogram(e) FROM UNNEST(regexp_extract_all(x, '.')) u(e))
     -> FROM (VALUES 'abccccdaee') t(x);
Query 20201211_083504_00003_be637 failed: line 1:9: Given correlated subquery is not supported
```

**Comment by martint** [comment]
More generally, this is about supporting correlated scalar queries that only reference values from the outer row, but don’t involve any additional table (so it’s not possible to de correlate via a traditional means involving joins)

It requires being able to actually execute the correlated subquery, similar to how “index joins” are planned and executed.

@erichwang, can you jot down the result from your investigation on what other pieces are needed to be able to pull this off so that we don’t forget?

**Comment by martint** [comment]
This is related to https://github.com/prestosql/presto/issues/12#issuecomment-534757849

**Comment by findepi** [comment]
> but don’t involve any additional table (so it’s not possible to de correlate via a traditional means involving joins)

I think this is possible to decorrelate today, at least for my example:

- assign unique id
- unnest
- aggregate, grouping by the unique id

**Comment by martint** [comment]
Yes, good point, you're right. I was thinking of the general case, which includes any table-generating construct in the FROM clause of the inner query. It might still be possible to decorrelate that by assigning a unique id (this is something we were contemplating, but at the execution level).
